src/exchange/moderators/summarizer.py

```diff
@@ -1,135 +1,52 @@
-from typing import List, Literal, Optional, Tuple, Type, Union
+from typing import Type
 
 from exchange import Message
-from exchange.checkpoint import Checkpoint
-from exchange.moderators import ContextTruncate, Moderator, PassiveModerator
+from exchange.moderators import ContextTruncate, PassiveModerator
 
-MAX_TOKENS = 112000
-SUMMARIZATION_OFFSET = 40000  # Keep a max of this many tokens
+# currently this is the point at which we start to summarize, so
+# so once we get to this token size the token count will exceed this
+# by a little bit
+MAX_TOKENS = 70000
```

Comment:
Same comment as for Truncate. We should start summarizing at a higher token count, but also having a knob to summarize a fraction (or offset) of the messages

Reply:
should ie also be per provider? claude has a much higher context window for example

Reply:
yea, definitely per provider. We need a way to set these in the profiles (on a backlog) so users can set these values directly instead of using the defaults.

Reply:
Agreed, we need to add two new things for providers: 
- `max_tokens` (variable)
- `is_allowed_to_call_llm` (method)

I've annotated both of these areas in the code with TODOs